lib/responses.js

```diff
@@ -408,6 +408,66 @@ function expectJSON(fn, res) {
   }).catch(handleError(res));
 }
 
+/**
+ * Checks the page size requested to make sure it's a Number and within the maximum.
+ * Returns the default or 0 if there's no default or size is not a Number.
+ * @param {Object} req
+ * @param {Object} res
+ * @returns {number}
+ */
+function checkPageSize(req, res) {
+  try {
+    let size;
+
+    // fall back to default
+    if (!req.query.size && process.env.CLAY_STORAGE_DEFAULT_PAGE_SIZE) {
+      size = Math.abs(Number(process.env.CLAY_STORAGE_DEFAULT_PAGE_SIZE));
+    } else if (req.query.size) {
+      // invalid size query param
+      if (isNaN(Math.abs(Number(req.query.size)))) {
+        throw Error('Client: Invalid size');
+      } else {
+        // valid size query
+        size = Math.abs(Number(req.query.size));
+      }
+    }
+
+    const max = Math.abs(Number(process.env.CLAY_STORAGE_MAX_PAGE_SIZE));
+
+    if (max) {
+      return size > max ? max : size;
+    } else return size;
+  } catch (error) {
+    handleError(res)(error);
+  }
+}
+
+/**
+ * Checks whether the previous value for a page request matches up with the uri,
+ * ie. all the ids returned should contain the request uri.
+ * @param {Object} req
+ * @param {Object} res
+ * @returns {string|undefined}
+ */
+function checkPrevious(req, res) {
+  try {
+    const prefix = req.uri || null;
+    const previous = req.query.prev || null;
+
+    if (previous) {
+      if (previous.includes(prefix)) {
+        return previous;
+      } else {
+        throw Error('Client: Invalid previous');
+      }
+    } else {
+      return;
+    }
+  } catch (error) {
+    handleError(res)(error);
+  }
+}
```

Comment:
```suggestion
function checkPrevious(req, res) {                                               
  const prefix = req.uri;                                                        
  const previous = req.query.prev;                                               
                                                                                 
  if (previous && prefix && !previous.includes(prefix)) {                        
    return handleError(res)(Error('Client: Invalid previous'));                  
  }                                                                              
                                                                                 
  return previous;                                                               
}
```

Do we need the `try/catch` here and the casting to `null` or would something like this work?

Reply:
Yep. This should work. Much cleaner obviously. Thanks for the suggestion.